Assets/Scripts/Utility/AssetInjection/MeshReplacement.cs

```diff
@@ -104,7 +104,13 @@ public static GameObject ImportCustomGameobject (uint modelID, Transform parent,
             go.name = GameObjectHelper.GetGoModelName(modelID) + " [Replacement]";
             go.transform.parent = parent;
             go.transform.position = matrix.GetColumn(3);
-            go.transform.rotation = GameObjectHelper.QuaternionFromMatrix(matrix);
+            go.transform.rotation = matrix.rotation;
+
+            //We can't simply apply scale, since we can't guarantee that custom models will have a defualt (1,1,1) scale
+            go.transform.localScale = new Vector3(go.transform.localScale.x * matrix.lossyScale.x,
+                                                    go.transform.localScale.y * matrix.lossyScale.y,
+                                                    go.transform.localScale.z * matrix.lossyScale.z);
```

Comment:
I think you can use `go.transform.localScale = Vector3.Scale(go.transform.localScale, matrix.lossyScale);`